src/app/credExplorer/state.js

```diff
@@ -0,0 +1,247 @@
+// @flow
+
+import deepEqual from "lodash.isequal";
+
+import * as NullUtil from "../../util/null";
+import {Graph} from "../../core/graph";
+import type {Repo} from "../../core/repo";
+import {type EdgeEvaluator} from "../../core/attribution/pagerank";
+import {
+  type PagerankNodeDecomposition,
+  type PagerankOptions,
+  pagerank,
+} from "../../core/attribution/pagerank";
+
+import type {DynamicPluginAdapter} from "../pluginAdapter";
+import {StaticPluginAdapter as GitAdapter} from "../../plugins/git/pluginAdapter";
+import {StaticPluginAdapter as GithubAdapter} from "../../plugins/github/pluginAdapter";
+
+/*
+  This models the UI states of the credExplorer/App as a state machine.
```

Comment:
You probably want to export the `initialState: () => AppState`.

Reply:
Done.